pandas/core/internals/construction.py

```diff
@@ -150,7 +150,7 @@ def init_ndarray(values, index, columns, dtype=None, copy=False):
 
         index, columns = _get_axes(len(values), 1, index, columns)
         return arrays_to_mgr([values], columns, index, columns, dtype=dtype)
-    elif is_extension_array_dtype(values):
+    elif is_extension_array_dtype(values) or is_extension_array_dtype(dtype):
```

Comment:
I'm not super well versed in this area but do you know what was actually throwing the error? Just wondering if there's a more generic fix to be made

Reply:
This fix is consistent with the handling of Categorical in the preceding condition.